src/framework/components/rigid-body/system.js

```diff
@@ -552,6 +560,211 @@ class RigidBodyComponentSystem extends ComponentSystem {
         return results;
     }
 
+    /**
+     * Perform a collision check on the world and return all entities the shape hits.
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {object} shape - The shape to use for collision.
+     * @param {number} shape.axis - The local space axis with which the capsule, cylinder or cone shape's length is aligned. 0 for X, 1 for Y and 2 for Z. Defaults to 1 (Y-axis).
+     * @param {Vec3} shape.halfExtents - The half-extents of the box in the x, y and z axes.
+     * @param {number} shape.height - The total height of the capsule, cylinder or cone from tip to tip.
+     * @param {string} shape.type - The type of shape to use. Available options are "box", "capsule", "cone", "cylinder" or "sphere". Defaults to "box".
+     * @param {number} shape.radius - The radius of the sphere, capsule, cylinder or cone.
+     * @param {Vec3} [position] - The world space position for the shape to be.
+     * @param {Vec3} [rotation] - The world space rotation for the shape to have.
+     *
+     * @returns {import('../../entity.js').Entity[]} An array of shapecast hit results (0 length if there were no hits).
+     */
+    shapeCast(shape, position, rotation) {
+        if (shape.type === 'capsule') {
+            return this.capsuleCast(shape.radius, shape.height, shape.axis, position, rotation);
+        } else if (shape.type === 'cone') {
+            return this.coneCast(shape.radius, shape.height, shape.axis, position, rotation);
+        } else if (shape.type === 'cylinder') {
+            return this.cylinderCast(shape.radius, shape.height, shape.axis, position, rotation);
+        } else if (shape.type === 'sphere') {
+            return this.sphereCast(shape.radius, position, rotation);
+        }
+
+        return this.boxCast(shape.halfExtents, position, rotation);
+    }
```

Comment:
switch case could be used here IMHO

Reply:
```js
switch (shape.type) {
  case 'capsule':
    return this.capsuleCast(shape.radius, shape.height, shape.axis, position, rotation);
  case 'cone':
    return this.coneCast(shape.radius, shape.height, shape.axis, position, rotation);
  case 'cylinder':
    return this.cylinderCast(shape.radius, shape.height, shape.axis, position, rotation);
  case 'sphere':
    return this.sphereCast(shape.radius, position, rotation);
  default:
    return this.boxCast(shape.halfExtents, position, rotation);
}

```

Reply:
Ah yeah, after test on mesurethat.com it seems that it's slightly more performant, that's updated!